inc/LocalProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Local media provider
+ *
+ * @package asset-manager-framework
+ */
+
+declare( strict_types=1 );
+
+namespace AssetManagerFramework;
+
+class LocalProvider extends Provider {
+	public function get_id() : string {
+		return 'local';
+	}
+
+	public function get_name() : string {
+		return __( 'Local Media', 'asset-manager-framework' );
+	}
+
+	protected function request( array $args ) : MediaList {
+		wp_ajax_query_attachments();
+		return new MediaList();
```

Comment:
So this is a bit of a special case - it only works because `wp_ajax_query_attachments()` is the default ajax handler for the media library and as such will call one of the `wp_send_json_*` functions - the `return new MediaList()` will never be executed.

Reply:
The special handling could be moved to the function in `namespace.php` that handles fetching results from the providers but I thought we could keep that separation clean. I can at least add a comment though as you say.

Reply:
Ah, you are right. I didn't realize that `wp_ajax_query_attachments` is actually the full AJAX endpoint, `exit`-ing the request. I somehow thought this was executed as part of the AJAX handling, but missed that it's the real thing.

Adding a quick comment would be enough/good then, yes. Thanks for clarifying. 🙂